spring-integration-core/src/main/java/org/springframework/integration/dsl/ConsumerEndpointSpec.java

```diff
@@ -170,15 +173,19 @@ public S requiresReply(boolean requiresReply) {
 	/**
 	 * @param sendTimeout the send timeout.
 	 * @return the endpoint spec.
-	 * @see AbstractReplyProducingMessageHandler#setSendTimeout(long)
+	 * @see AbstractMessageProducingHandler#setSendTimeout(long)
 	 */
 	public S sendTimeout(long sendTimeout) {
 		assertHandler();
-		if (this.handler instanceof AbstractReplyProducingMessageHandler) {
-			((AbstractReplyProducingMessageHandler) this.handler).setSendTimeout(sendTimeout);
+		if (this.handler instanceof AbstractMessageProducingHandler) {
+			((AbstractMessageProducingHandler) this.handler).setSendTimeout(sendTimeout);
+		}
+		else if (this.handler instanceof AbstractMessageRouter) {
+			// This should probably go on the RouterSpec, but we put it here for consistency
```

Comment:
I see your concern for discussion, but IMO you have done the proper decision.
This option must be here (it is here already anyway) just because it really will be a mess to have it on the `RouterSpec` as well:
```
.route("true", r -> r.sendTimeout(100), e -> e.sendTimeout(100))
```